### ISSUE TITLE
Feature proposal: add "components" option to attributifyPreset

### ISSUE THREAD

**Issue by smashercosmo** [unocss/unocss#4317]
### Clear and concise description of the problem

Many design systems have such base components as Flex, Grid, Text, Box, Stack and so on. These components usually accept props and values that look exactly like (or very similar to) css.
```tsx
<Text weight="bold" size="sm">Hello</Text>
<Flex align="center" gap="2rem"><div>One</div><div>Two</div></Flex>
<Box width="1rem" height="1rem">Square</Box>
```
And this is the perfect case for the attributify preset. But, there could be other components in the project, which accept similar props (like for example ```<Button size="sm">Click me</Button>```) and which shouldn't be processed by the preset. Currently, the only way to circumvent it is to provide a prefix for attributify preset, but it makes attributes more verbose.

### Suggested solution

It would be nice if you could just specify a list of components or html tags that should be processed by the preset. And all other components will be ignored and won't be included in the extraction process. It could look something like this:
```
export default defineConfig({
  presets: [
    presetAttributify({
      components: ["Flex", "Box", "Grid", "Text"]
    }),
    // ...
  ],
})
```

### Alternative

The alternative is to just use prefix option

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Read the `README.md` of using the package.
- [X] Already used the [Interactive Docs](https://unocss.dev/interactive/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by zyyv** [comment]
It sounds like https://unocss.dev/presets/tagify

**Comment by smashercosmo** [comment]
I'm not sure that, what I described, can be achieved with tagify preset. Basically, I just need a way to specify which components should be processed by the attributify preset. I played a little bit with the attributify source and seems like this feature can be implemented in just two lines of code. Can submit a PR if the proposed feature is something that you might consider worth having.

**Comment by zyyv** [comment]
You can send pr and test first, then install the temporary package from `pkg.pr.new`, you can test it in your project to see if it works well/
BTW, I prefer it to be implemented in `tagify`. 🤔

**Comment by smashercosmo** [comment]
How do you see it implemented in tagify? My real-life use case is that I have `Text` component, which has some internal logic, like, for example, support for the left-hand side ellipsis (here is the shortened version of the code):
```tsx
import { type ReactNode } from "react";
import { css, cx } from "@emotion/css";

type Props = {
  children: ReactNode;
  truncate?: boolean | "start" | "end";
};

const truncateRootStyles = css`
  display: inline-grid;
`;

const truncateContentStyles = css`
  text-overflow: ellipsis;
  overflow: hidden;
  text-wrap: nowrap;
`;

const truncateStartStyles = css`
  direction: rtl;
  text-align: start;
`;

export function Text({
  truncate,
}: Props) {
  const content = truncate ? (
    <span
      className={cx(truncateContentStyles, {
        [truncateStartStyles]: truncate === "start",
      })}
      title={typeof children === "string" ? children : undefined}
    >
      {truncate === "start" && (
        /* This is a weird trick. This character prevents special symbols like "-", "!", "?" etc. */
        /* to be moved to the right when direction is set to "rtl" */
        <span className={hiddenCharacterStyles} aria-hidden="true">
          x
        </span>
      )}
      {children}
    </span>
  ) : (
    children
  );

  return (
    <span
      className={cx({
        [truncateRootStyles]: Boolean(truncate),
      })}
    >
      {content}
    </span>
  );
}
```

And I want this component to have both the internal logic and the ability to be processed by the attributify preset. And at the same time I don't want other components to be processed by the attributify preset.

**Comment by zhangmo8** [comment]
But for css, it should not contain more logic. If you really need labeled attributes and related logic, maybe you need a semantic component library. For example: [chakra](https://www.chakra-ui.com/docs/components/flex) or other

**Comment by iamyuu** [comment]
I think the proposal is to add new options to define "variant".

###### Proposal:

```ts
presetTagify({
	component: {
		// ↓ attribute key
		text: {
			// ↓ variant key
			variant: {
				// ↓ variant value
				heading: 'text-2xl font-bold', // ← the compiler will remove the variant key and add this class to the element
				body: 'text-base font-normal',
				label: 'text-sm font-medium',
				caption: 'text-xs font-light',
			}
		},
		button: {
			size: {
				sm: 'text-sm',
				md: 'text-base',
				lg: 'px-4 py-3 text-lg'
			}
		}
	}
})
```

###### Usage:

```html
<text variant="caption">Heading</text>
<button size="lg">Button</button>
```

###### Result:

```html
<div class="text-xs font-light">Heading</div>
<button class="text-base px-4 py-3 text-lg">Button</button>
```

Inspired by [Stitches](https://stitches.dev/), [CVA](https://github.com/joe-bell/cva), [Tailwind Variants](https://www.tailwind-variants.org).

**Comment by smashercosmo** [comment]
No, it's not. The proposal is to make it possible to define a list of certain components or tags to be processed by unocss. Components or tags not included in the list will be left intact.
